open-company/open-company-auth
# Recipient validation

## Review after #111 

#### Card: https://trello.com/c/YBJz95A6

#### Related PRs (review together):
- [Web](https://github.com/open-company/open-company-web/pull/1183)
- [Lib](https://github.com/open-company/open-company-lib/pull/89)

#### Description of the change:
Use sparkpost.com to prevent further problems with our SES account. We are going to check the email of every user being inserted in our system when the record is created. If the email result fake (see code for details on what is a fake email) we drop the user and return an error to the client.

To test:
- sign in with your stoatlabs Slack account
- try inviting yourself with an alias email, like calvin+teststoat@stoatlabs.com
- now invite a fake user like: asdjsajduj@sdiauhdbasiouhduasibhfuiashbfuohouifhaud.com
- [ ] did it fail? Good

This should be it, please drop me a note when this is merged since I need to do a bit of cleanup in the infra repo (deployed branch mostly).
